Money/Shared/Money.swift

```diff
@@ -55,9 +55,52 @@ public protocol MoneyType: DecimalNumberType, ValueCoding {
 
 public extension MoneyType where DecimalStorageType == NSDecimalNumber {
 
+    /// Convenience access to the "amount" as an NSDecimalNumber.
     var amount: DecimalStorageType {
         return storage
     }
+
+    /**
+
+     ### Formatted String
+
+     This function will format the Money type into a string for 
+     the current locale.
+
+     For custom currencies which define their own currency code,
+     create a lazy static `NSNumberFormatter`. Set the following
+     properties on it: `currencySymbol`, `internationalCurrencySymbol`
+     `currencyGroupingSeparator` and `currencyDecimalSeparator`, as they
+     will be used when formatting the string. Feel free to fall back to
+     the current locale's values for any of these to maintain
+     natural looking formatting. See the example project for more.
+
+     - parameter style: the `NSNumberFormatterStyle` to use.
+     - returns: a localized and formatted string for the money amount.
+     */
+    func formattedWithStyle(style: NSNumberFormatterStyle) -> String {
+        return Currency.formattedWithStyle(style, forLocaleId: NSLocale.currentLocale().localeIdentifier)(amount)
+    }
+
+    /**
+
+     ### Formatted String for specific Locale
+
+     This function will format the Money type into a string suitable
+     for a specific local. It accepts an parameter for the
+     style `NSNumberFormatterStyle`. Note that iOS 9 and OS X 10.11
+     added new styles which are relevant for currency.
+
+     These are `.CurrencyISOCodeStyle`, `.CurrencyPluralStyle`, and
+     `.CurrencyAccountingStyle`.
+
+     - parameter style: the `NSNumberFormatterStyle` to use.
+     - parameter locale: a `Locale` value
+     - returns: a localized and formatted string for the money amount.
+     */
+    func formattedWithStyle(style: NSNumberFormatterStyle, forLocale locale: Locale) -> String {
+        return Currency.formattedWithStyle(style, forLocale: locale)(amount)
+    }
```

Comment:
These changes represent the API for localized string formatting that I'm going with. At the moment, the only thing left to sort out is to update the unit tests.

In usage, this works pretty much as you would expect. For normal circumstances, i.e. current device locale - use `.description` and `formattedWithStyle`. Like this:

```swift
print("current locale id: \(NSLocale.currentLocale().localeIdentifier)")
let money: Money = 99.99
print("The amount is: \(money)")
print("She has \(money.formattedWithStyle(.CurrencyPluralStyle))")
```
This will print out whatever the device's local currency is, using the device's current locale. 

For an American in Russia:
>current locale id: en_RU
>The amount is: 99,99 RUB
>She has 99,99 Russian roubles

For a Spaniard in Russia you get:
>current locale id: es_RU
>The amount is: 99,99 RUB
>She has 99,99 rublos rusos

Lets say you want to use, whatever the local currency is, but you want to format strings for a specific local. We can change the description to use code like this:

```swift
print("The amount is: \(money.formattedWithStyle(.CurrencyStyle, forLocale: .German(.Switzerland)))")
print("She has \(money.formattedWithStyle(.CurrencyPluralStyle, forLocale: .Chinese(.China)))")
```

For the American, you get:
>current locale id: en_RU
>The amount is: RUB 99.99
>She has 99.99俄罗斯卢布

Which is the same as for the Spaniard, (as expected):
>current locale id: es_RU
>The amount is: RUB 99.99
>She has 99.99俄罗斯卢布

There are a couple of limitations with this.

Essentially any custom currencies will not get the fantastic localization support here for the `.CurrencyPluralStyle`. I don't think it's possible to provide `NSNumberFormatter` with the translations to do this. So, for Bitcoin (`BTC`) using this style will use whatever the name of the local currency is.

Likewise, if you have your own in-app currency, avoid the `.CurrencyPluralStyle` style.

The `Locale` parameter is an auto-generated enum using US English case names. Think of it like `.Language(.CountryWhichSpeaksThatLanguage)`. In some cases, there is only one country which speaks that language, in which case there is no associated value.